Wrong line position when using non-usize type in allocation

### Zig Version

0.11.0-dev.4296+7e25fb4a4

### Steps to Reproduce and Observed Output

```zig
const std = @import("std");
const Allocator = std.mem.Allocator;

const Configuration = struct {
    dimension: usize,
};

const State = struct {
    const Self = @This();

    test_buffer: []f32,
    other_buffer: []f32,

    fn init(allocator: Allocator, config: *const Configuration) !Self {
        var xxX: i32 = 0;
        return Self{
            .test_buffer = try allocator.alloc(f32, xxX),
            .other_buffer = try allocator.alloc(f32, config.dimension),
        };
    }
};

pub fn main() !void {
    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
    defer arena.deinit();
    const allocator = arena.allocator();

    const config = Configuration{
        .dimension = 100,
    };

    var state = try State.init(allocator, &config);
    _ = state;
}
```

zig run example.zig
```sh
❯ zig run example.zig
example.zig:2:19: error: expected type 'usize', found 'i32'
const Allocator = std.mem.Allocator;
                  ^~~
example.zig:2:19: note: unsigned 64-bit int cannot represent all 
possible signed 32-bit values
referenced by:
    main: example.zig:31:26
    callMain: /home/cgbur/zig/0.11.0-dev.4296+7e25fb4a4/files/lib
/std/start.zig:615:32
    remaining reference traces hidden; use '-freference-trace' to
 see all reference traces
```

In a project I am working on with fairly large structs it is pointing at a field in a completely different struct nearly a hundred lines above.

### Expected Output

Something about
```
.test_buffer = try allocator.alloc(f32, xxX),
                                        ^^^^
```
highlighting the xxX.

--- a/src/Sema.zig
+++ b/src/Sema.zig
@@ -7622,7 +7622,7 @@ fn instantiateGenericCall(
             else if (call_src == .node_offset) .{ .call_arg = .{
                 .decl = block.src_decl,
                 .call_node_offset = call_src.node_offset.x,
-                .arg_index = @intCast(total_i),
+                .arg_index = @intCast(total_i - @intFromBool(bound_arg_src != null)),
             } } else .unneeded;
 
             const comptime_arg = callee.comptime_args.get(ip)[total_i];
@@ -9341,17 +9341,6 @@ fn zirParam(
         // Even though a comptime argument is provided, the generic function wants to treat
         // this as a runtime parameter.
         assert(sema.inst_map.remove(inst));
-    }
-
-    if (sema.generic_owner != .none) {
-        if (try sema.typeHasOnePossibleValue(param_ty)) |opv| {
-            // In this case we are instantiating a generic function call with a non-comptime
-            // non-anytype parameter that ended up being a one-possible-type.
-            // We don't want the parameter to be part of the instantiated function type.
-            sema.inst_map.putAssumeCapacity(inst, Air.internedToRef(opv.toIntern()));
-            sema.comptime_args[param_index] = opv.toIntern();
-            return;
-        }
     }
 
     try block.params.append(sema.arena, .{

--- a/test/cases/compile_errors/generic_method_call_invalid_coercion.zig
+++ b/test/cases/compile_errors/generic_method_call_invalid_coercion.zig
@@ -1,0 +1,21 @@
+export fn callBoolMethod() void {
+    const s = S{};
+    s.boolMethod({});
+}
+
+export fn callVoidMethod() void {
+    const s = S{};
+    s.voidMethod(false);
+}
+
+const S = struct {
+    fn boolMethod(comptime _: @This(), _: bool) void {}
+    fn voidMethod(comptime _: @This(), _: void) void {}
+};
+
+// error
+// backend=stage2
+// target=native
+//
+// :3:18: error: expected type 'bool', found 'void'
+// :8:18: error: expected type 'void', found 'bool'

